lib/i18n/server.ts

```diff
@@ -0,0 +1,11 @@
+import { cache } from "react";
+import { cookies, headers } from "next/headers";
+import { cookieName, fallbackLng } from "./settings";
+
+export const getLocale = cache(() => {
```

Comment:
utility for getting the locale on the server outside of the `[locale]` directory context